network/test/middleware_test.go

```
@@ -525,6 +531,329 @@ func (m *MiddlewareTestSuite) TestUnsubscribe() {
+			// because we are using the default overlay a random identity is returned
+			// so we wait until we get the arguments before checking them.
+			expectedViolation.Identity = actualViolation.Identity
+			require.Equal(m.T(), expectedViolation, actualViolation)
+		})
+
+		defer slashingViolationsConsumer.AssertExpectations(m.T())
+
+		ids, libP2PNodes, _ := GenerateIDs(m.T(), m.logger, 1)
+		mws, providers := GenerateMiddlewares(m.T(), m.logger, ids, libP2PNodes, unittest.NetworkCodec(), slashingViolationsConsumer)
+		require.Len(m.T(), ids, 1)
+		require.Len(m.T(), providers, 1)
+		require.Len(m.T(), mws, 1)
+		newId := ids[0]
+		newMw := mws[0]
+
+		overlay := m.createOverlay(providers[0])
+		overlay.On("Receive",
+			m.ids[0].NodeID,
+			mock.AnythingOfType("*message.Message"),
+		).Return(nil)
+		newMw.SetOverlay(overlay)
+
+		ctx, cancel := context.WithCancel(m.mwCtx)
+
+		mwCtx, _ := irrecoverable.WithSignaler(ctx)
+		newMw.Start(mwCtx)
+
+		idList := flow.IdentityList(append(m.ids, newId))
+
+		// needed to enable ID translation
+		m.providers[0].SetIdentities(idList)
+
+		//NOTE: set the channel ID in the test message to a channel that TestMessage(s) is not allowed to be sent on ConsensusCommittee
+		msg, _ := createMessage(m.ids[0].NodeID, newId.NodeID, "hello")
+		msg.ChannelID = channels.ConsensusCommittee.String()
+
+		// update the addresses
+		m.mws[0].UpdateNodeAddresses()
+
+		// send message via unicast
+		err = m.mws[0].SendDirect(msg, newId.NodeID)
+		require.NoError(m.T(), err)
+
+		// wait for slashing violations consumer mock to invoke run func and close ch if expected method call happens
+		unittest.RequireCloseBefore(m.T(), ch, 100*time.Millisecond, "slashing violations consumer mock not invoked on time")
+
+		// cleanup newMW resources
+		cancel()
+		unittest.RequireCloseBefore(m.T(), newMw.Done(), 100*time.Millisecond, "could not stop middleware on time")
+	})
+
+	m.Run("unknown message code", func() {
```

can you add a similar test where the message code is overwritten with a different but legitimate message code?

---

This commit contains an update to the codec so that an error is returned when an message cannot be unmarshalled rather than failing silently. It also adds corresponding sentinel and slashing violation consumer event. 
https://github.com/onflow/flow-go/pull/2832/commits/7794d5cd2c446ebd0f4a6988289b9a4e07d77c97